docs/basic-features/data-fetching/client-side.md

```diff
@@ -0,0 +1,68 @@
+---
+description: 'Learn how to use SWR, a data fetching React hook that handles caching, revalidation, focus tracking, refetching on interval and more.'
+---
+
+# Client-side data fetching
+
+Client-side data fetching is useful when your page doesn't require SEO indexing, when you don't need to pre-render your data, or when your pages content needs to update frequently. Unlike the server-side rendering APIs, you can use client side data fetching at the component level.
+
+If done at the page level, the data is fetched at runtime, and the contents of the page updated as the data changes. When used at the component level, the data is fetched at the time of the component mount, and the contents of the component are updated as the data changes.
+
+It's important to note that using client-side data fetching can affect the performance of your application and the load speed of your pages. This is because the data fetching is done at the time of the component or pages mount, and the data is not cached.
+
+## Client-side data fetching with useEffect
+
+The following example shows how you can fetch data on the client side using the useEffect hook.
+
+```jsx
+function Profile() {
+  const [data, setData] = useState(null)
+  const [isLoading, setLoading] = useState(false)
+
+  useEffect(() => {
+    setLoading(true)
+    fetch('api/profile-data')
+      .then((res) => res.json())
+      .then((data) => {
+        setData(data)
+        setLoading(false)
+      })
+  }, [])
+
+  if (isLoading) return <p>Loading...</p>
+  if (!profileData) return <p>No profile data</p>
+
+  return (
+    <div>
+      <h1>{data.name}</h1>
+      <p>{data.bio}</p>
+    </div>
+  )
+}
+```
+
+# Client-side data fetching with SWR
+
+The team behind Next.js has created a React hook for data fetching called [**SWR**](https://swr.vercel.app/). It is **highly recommend** if you are fetching data on the client side. It handles caching, revalidation, focus tracking, refetching on interval, and more.
+
+Using the same example as above, we can now use SWR to fetch the profile data. SWR will automatically cache the data for us, and will revalidate the data if it is stale.
+
+For more information on using SWR, check out the [SWR docs](https://swr.vercel.app/docs).
+
+```jsx
+import useSWR from 'swr'
+
+function Profile() {
+  const { data, error } = useSWR('/api/profile-data', fetch)
```

Comment:
```suggestion
  const { data, error } = useSWR('/api/profile-data', fetch)
```

This will need a custom fetcher like this:
```js
const fetcher = (...args) => fetch(...args).then(res => res.json())
```
https://swr.vercel.app/docs/getting-started#quick-start